src/core/src/pattern/op/pattern.cpp

```diff
@@ -187,19 +188,40 @@ class AttributeMatchingVisitor : public ov::AttributeVisitor {
 public:
     explicit AttributeMatchingVisitor(const Attributes& expected_attrs)
         : ov::AttributeVisitor(),
-          m_expected_attrs{expected_attrs} {}
+          m_expected_attrs{expected_attrs} {
+        OPENVINO_ASSERT(!expected_attrs.empty(), "Please remove trivial attribute matching check");
+        for (auto& [name, expected_value] : expected_attrs)
+            m_matched_attributes[name] = false;
+    }
 
     void on_adapter(const std::string& name, ValueAccessor<void>& adapter) override {
-        if (m_attrs_match && m_expected_attrs.count(name)) {
+        if (m_expected_attrs.count(name)) {
             try {
                 const auto& node_attribute = adapter.get_as_any();
-                const auto& expected_attribute = m_expected_attrs.at(name);
+                const auto& attribute_type_id = node_attribute.type_info();
+
+                auto expected_attribute = m_expected_attrs.at(name);
+                const auto& expected_type_id = expected_attribute.type_info();
+
+                if (attribute_type_id != expected_type_id) {  // type conversions for developer convenience
+                    if (attribute_type_id == typeid(int64_t) && expected_type_id == typeid(int))
+                        expected_attribute = ov::Any(static_cast<int64_t>(expected_attribute.as<int>()));
+                    if (attribute_type_id == typeid(std::string) && expected_type_id == typeid(const char*))
+                        expected_attribute = ov::Any(std::string(expected_attribute.as<const char*>()));
+                    if (attribute_type_id == typeid(bool) && expected_type_id == typeid(int))
+                        expected_attribute = ov::Any(static_cast<bool>(expected_attribute.as<int>()));
+                    if (attribute_type_id == typeid(double) && expected_type_id == typeid(int))
+                        expected_attribute = ov::Any(static_cast<double>(expected_attribute.as<int>()));
+                    if (attribute_type_id == typeid(double) && expected_type_id == typeid(float))
+                        expected_attribute = ov::Any(static_cast<double>(expected_attribute.as<float>()));
```

Comment:
```suggestion
                        expected_attribute = ov::Any((expected_attribute.as<double>()));
```
If `expected_attribute` is ov::Any the and stores numerical value should be possible to convert to other numerical value.

Reply:
For the numerical values the conversion has been added recently see [test as example](https://github.com/praasz/openvino/blob/131bcad397f2b549637f56eab527a4a2320d5af4/src/core/tests/any.cpp#L759)
but this is only detail.